syntax = "proto3";

package contatos;

service ContatoService {
    rpc AdicionarContato (ContatoRequest) returns (ContatoResponse);
    rpc ConsultarContato (ConsultaRequest) returns (ContatoResponse);
    rpc ListarContatos (Empty) returns (ListaContatosResponse);
}

message Empty {}

enum TipoTelefone {
    MOVEL = 0;
    FIXO = 1;
    TIPO_COMERCIAL = 2;
}

enum CategoriaContato {
    FAMILIAR = 0;
    PESSOAL = 1;
    CATEGORIA_COMERCIAL = 2;
}

message Telefone {
    string numero = 1;
    TipoTelefone tipo = 2;
}

message Contato {
    string nome = 1;
    repeated Telefone telefones = 2;
    CategoriaContato categoria = 3;
}

message ContatoRequest {
    Contato contato = 1;
}

message ConsultaRequest {
    string nome = 1;
}

message ContatoResponse {
    Contato contato = 1;
    string mensagem = 2;
}

message ListaContatosResponse {
    repeated Contato contatos = 1;
}
